src/scripts/main.js

```diff
@@ -1,5 +1,40 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = wall.querySelector('.spider');
+
+document.addEventListener('click', (event) => {
+  const onWall = event.target.closest('.wall');
+
+  if (onWall) {
+    const x = event.clientX;
+    const y = event.clientY;
+    const styleWall = getComputedStyle(wall);
+    const rect = wall.getBoundingClientRect();
+    const borderWidth = parseFloat(styleWall.borderWidth);
+    const minLeft = rect.left + borderWidth + spider.offsetWidth / 2;
+    const minTop = rect.top + borderWidth + spider.offsetHeight / 2;
+    const maxLeft = rect.right - borderWidth - spider.offsetHeight / 2;
```

Comment:
There is a mistake in the calculation of `maxLeft`. It should subtract `spider.offsetWidth / 2` instead of `spider.offsetHeight / 2` to correctly calculate the maximum left position.